lib/Jit/LLILCJit.cpp

```diff
@@ -641,3 +641,12 @@ void ObjectLoadListener::getRelocationTypeAndAddend(uint8_t *FixupAddress,
     llvm_unreachable("Unknown reloc type.");
   }
 }
+
+unsigned LLILCJit::getMaxIntrinsicSIMDVectorLength(DWORD CpuCompileFlags) {
+  return GenIR::getPreferIntrinsicSIMDVectorLength();
+}
+
+bool LLILCJitContext::isAltJit() {
+  JitOptions JitOptions(*this);
+  return JitOptions.IsAltJit;
```

Comment:
Yes, you're right. In that case, we should expose a new flag on `Options` (see include/Reader/options.h) informed by JitOptions.IsAltJit that indicates whether or not to use `GenIR`'s preferred SIMD vector size, then drive the logic in `GenIR::getMaxIntrinsicSIMDVectorLength` off of that flag.